### ISSUE TITLE
Can't use an equals sign "=" in a serverless.yml variable default

### ISSUE THREAD

**Issue by jacobevans** [serverless/serverless#5136]
# This is a Bug Report

## Description

When using `${var, default}` syntax, if the default is a string with an "=" in it, the whole `${}` construct will just be returned literally.

For example (this is in env variable section, for ease of demonstration) but it seems to apply anywhere)
serverless.yml:

    provider: 
      environment:
        TEST1: ${self:does_not_exist, "This_is_fine"}
        TEST2: ${self:does_not_exist, "this=not_fine"}

shell:

    $ sls package -s dev
    $ cat .serverless/cloudformation-template-update-stack.json  | grep TEST
            "TEST1": "This_is_fine",
            "TEST2": "${self:does_not_exist, \"this=not_fine\"}",

I would expect `TEST2` to have a value of `this=not_fine`

## Additional Data

* Serverless 1.28.0
* Mac OSX

**Comment by ronkorving** [comment]
Interestingly, [the regular expression](https://github.com/serverless/serverless/blob/cc55147ee25b96765ffa8b5f91563712a43c55f3/lib/classes/Variables.js#L53) used to detect constant strings, has a [`\1`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#special-backreference) reference in it, unlike all other regular expressions.

I wonder if this has something to do with it. If not used properly, it can cause weird side-effects. For example:

```js
const stringRefSyntax = RegExp(/(?:('|").*?\1)/g);
stringRefSyntax.exec('"this is fine"');
// [""this is fine"", """, index: 0, input: ""this is fine"", groups: undefined]
stringRefSyntax.exec('"this is fine"'); // same string! 2nd execution
// null
```

**Comment by mscharp** [comment]
Has anyone taken a look at this? Anyway to get this fix prioritized?

**Comment by dschep** [comment]
Confirmed the bug in [dschep-bug-repos/sls-5136](https://github.com/dschep-bug-repos/sls-5136), and even tried a few work arounds in hopes of a quick temporary fix, but to no avail 😢  

The regex using `\1` makes sense AFAICT because you want a string starting with `'` to end with `'` and similarly for `"`.

**Comment by mscharp** [comment]
I am no regex guru, but it the regex event the issue?  Putting that regex into an online regex tester and "this=not_fine", shows that it matches the whole string... I would assume then that the regex isn't the problem?

**Comment by dschep** [comment]
Agreed, @mscharp. Just addressing @ronkorving's thought about it.

**Comment by medikoo** [comment]
Fixed with #8987
